### PR TITLE
allow @Service and @Reference to merge attributes form annotations in lower levels of the annotation hierachy.

### PR DESCRIPTION
**Pull request by dhchao11**
## What is the purpose of the change
the related issue is : 
https://github.com/apache/incubator-dubbo/issues/4075

## Brief changelog

allow @Service and @Reference to merge attributes form annotations in lower levels of the annotation hierachy.

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/incubator-dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by dhchao11** [comment]
please review, tks

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+package org.apache.dubbo.config.spring.annotation.merged;
+
+
+import org.apache.dubbo.config.annotation.Reference;
+import org.springframework.core.annotation.AliasFor;
+
+import java.lang.annotation.*;
+
+/**
+ * @author dhchao11
```
Comment (CrazyHZM) [inline dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java]: Remove author information.

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.config.spring.annotation.merged;
+
+
+import org.apache.dubbo.config.annotation.Reference;
+import org.springframework.core.annotation.AliasFor;
+
+import java.lang.annotation.*;
```
Comment (CrazyHZM) [inline dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java]: Do not introduce all dependencies.

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedService.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.dubbo.config.spring.annotation.merged;
+
+
+import org.apache.dubbo.config.annotation.Service;
+import org.springframework.core.annotation.AliasFor;
+
+import java.lang.annotation.*;
```
Comment (CrazyHZM) [inline dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedService.java]: The same as above

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedService.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * @author dhchao11
```
Comment (CrazyHZM) [inline dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedService.java]: The same as above.

File: dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java (hunk 1 above)
Comment (dhchao11) [reply dubbo-config/dubbo-config-spring/src/test/java/org/apache/dubbo/config/spring/annotation/merged/MergedReference.java]: i have fixed it.

**Comment by mercyblitz** [comment]
Thanks for your PR, approved and merged

**Comment by dhchao11** [comment]
my pleasure, thank you all

**Pull request merged**